### PR TITLE
🌿 Add integration tests

### PR DESCRIPTION
**Pull request by amckinney**
This adds a series of integration tests to match those found in Seam's other SDKs (e.g. https://github.com/seamapi/java). A couple tests are commented out because the behavior is not yet supported by [`fake-seam-connect`](https://github.com/seamapi/fake-seam-connect).

## Which resources are tested
- [x] Access Codes
- [x] Action Attempts
- [x] Connected Accounts
- [x] Connected Webviews
- [x] Devices
- [x] Events
- [x] Noise Thresholds
- [x] Thermostats

## How does it work
* Each resource is split out into its own test file (e.g. `test/integration/access_codes_test.go`).
* Each resource test invokes `fake-seam-connect` on startup.
* Each test uses the generated SDK and asserts conditions.

## Gaps

There are a couple gaps in the test because `fake-seam-connect` does not support a subset of Seam's API. In particular,
* Connecting accounts via email does not work.
* Updating access codes with `start_at` and `ends_at` does not work.
* Updating thermostats does not work.

However, these gaps are relatively minor as the most important resources/tests work as expected.

### CODE REVIEW

File: test/integration/access_codes_test.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+		deviceIds = append(deviceIds, accessCode.DeviceId)
+	}
+	createdAccessCodes, err := seam.AccessCodes.CreateMultiple(
+		ctx,
+		&seamgo.AccessCodesCreateMultipleRequest{
+			DeviceIds: deviceIds,
+		},
+	)
+	require.NoError(t, err)
+	assert.GreaterOrEqual(t, len(createdAccessCodes), 1)
```
Comment (Neiva07) [inline test/integration/access_codes_test.go:153]: can you use the exact assertion?

File: test/integration/connect_webviews_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	assert.NotEmpty(t, newWebview.ConnectWebviewId)
+
+	gotWebview, err = seam.ConnectWebviews.Get(
+		ctx,
+		&seamgo.ConnectWebviewsGetRequest{
+			ConnectWebviewId: newWebview.ConnectWebviewId,
+		},
+	)
+	require.NoError(t, err)
+	assert.GreaterOrEqual(t, len(gotWebview.AcceptedProviders), 1)
```
Comment (Neiva07) [inline test/integration/connect_webviews_test.go:55]: same here

File: test/integration/connected_accounts_test.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+	connectedAccounts, err = seam.ConnectedAccounts.List(ctx)
+	require.NoError(t, err)
+	assert.Len(t, connectedAccounts, 1)
+}
+
+func getTestConnectedAccount(t *testing.T, seam *seamclient.Client) *seamgo.ConnectedAccount {
+	connectedAccounts, err := seam.ConnectedAccounts.List(context.Background())
+	require.NoError(t, err)
+	assert.GreaterOrEqual(t, len(connectedAccounts), 1)
```
Comment (Neiva07) [inline test/integration/connected_accounts_test.go:69]: same

**Review by Neiva07 — COMMENTED** [review]
lgtm, overall only a small request.

**Comment by amckinney** [comment]
@Neiva07 I adapted all the `GreaterOrEqual` checks to exact `Len` checks, but I think it's best to keep using `GreaterOrEqual` in the `getTestConnectedAccount` and `getTestDevice` helper functions. These helpers aren't actually meant to be testing the exact length of the response - the assertion is only there to guarantee they can select the first element from the returned list.

If a test author ever creates one of these resources in their test, they wouldn't be able to successfully use the helper function because the exact length check would no longer be true. But I agree it's much better to use exact length checks in the actual test logic. Let me know what you think!

**Review by rchodava — APPROVED** [review]

**Pull request merged**